docs/contents/dev-docs/testing.md

```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
+### IAM setup
 
-Then run:
-```
-make kind-cluster SERVICE=$SERVICE
-```
+In order for the ACK service controller to manage the S3 bucket, it needs an
+identity. In other words, it needs an IAM role that represents the ACK service
+controller towards the S3 service.
 
-This provisions a Kubernetes cluster using `kind`, builds a container image with
-the ACK service controller, and loads the container image into the `kind` cluster.
-It then installs the ACK service controller and related Kubernetes manifests into
-the `kind` cluster using `kustomize build | kubectl apply -f -`.
+In the following, we assume that you know your AWS account ID. Let's say it's 
+`1234567890121` (replace with your own in the following):
 
-Then, the above script runs a series of bash test scripts that call `kubectl`
-and the `aws` CLI tools to verify that custom resources of the type managed by
-the respective ACK service controller is created, updated and deleted
-appropriately (still TODO).
+```
+export ACCOUNT_ID=1234567890121
+```
 
-Finally, it will run E2E tests, which will create resources for the respective service in your AWS account in `us-west-2` region and 
-verify if the resource has successfully created. 
 
-The script deletes the `kind` cluster. You can prevent this last
-step from happening by passing the `-p` (for "preserve") flag to the
-`scripts/kind-build-test.sh` script or by `make kind-cluster-preserve SERVICE=ecr`.
+Next, define the name of the IAM role that will have the permission to manage
+S3 buckets on your behalf:
 
-!!! info 
-     For above functional testing, `generate_temp_creds` function under `scripts/lib/aws.sh` script will 
-     make `aws sts assume-role --role-session-arn $AWS_ROLE_ARN --role-session-name $TEMP_ROLE` API call 
-     to fetch `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN`. 
-     The duration of the session token is 900 seconds (15 minutes). These variables will be added as Environment variables to the controller (deployment). 
-
-## IAM setup for users to assume a role 
+```
+export ACK_TEST_IAM_ROLE=Admin-k8s
+```
 
-* To verify which IAM entity is making assume role API call, run `aws sts get-caller-identity` command:
+Now we need to verify the IAM entity (likely an IAM user) that is going to
+assume the IAM role `ACK_TEST_IAM_ROLE`, so execute:
 
 ```
 aws sts get-caller-identity
 ```
 
-* Check if the above returned IAM entity has necessary permissions to make assume role API call. 
-  The contents of the example-role-trust-policy.json file should be similar to this:    
-``` json
-{
-	"Version": "2012-10-17",
-	"Statement": {
-		"Effect": "Allow",
-		"Principal": {
-			"AWS": "arn:aws:iam::12345678970:user/kubernetes"
-		},
-		"Action": "sts:AssumeRole"
-	}
-}
-```
+From this above command's output, note the `Arn` field value, something along
+the lines of `arn:aws:iam::1234567890121:user/ausername` and store it in
+an environment variable called `ACK_PRINCIPAL_ARN`.
```

Comment:
How about `ACK_TEST_PRINCIPAL_ARN` to be consistent with the test role env var name?